modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/HorizontalCardTag.java

```diff
@@ -0,0 +1,383 @@
+			return _horizontalCard.isInteractive();
+		}
+
+		return _interactive;
+	}
+
+	public String getTitle() {
+		if ((_title == null) && (_horizontalCard != null)) {
+			return _horizontalCard.getTitle();
+		}
+
+		return _title;
+	}
+
+	public Boolean isDisabled() {
+		if (_disabled == null) {
+			if (_horizontalCard == null) {
+				return false;
+			}
+
+			return _horizontalCard.isDisabled();
+		}
+
+		return _disabled;
+	}
+
+	public Boolean isSelectable() {
+		if (_selectable == null) {
+			if (_horizontalCard == null) {
+				return false;
+			}
+
+			return _horizontalCard.isSelectable();
+		}
+
+		return _selectable;
+	}
+
+	public Boolean isSelected() {
+		if (_selected == null) {
+			if (_horizontalCard == null) {
+				return false;
+			}
+
+			return _horizontalCard.isSelected();
+		}
+
+		return _selected;
+	}
+
+	public void setActionDropdownItems(List<DropdownItem> actionDropdownItems) {
+		_actionDropdownItems = actionDropdownItems;
+	}
+
+	public void setData(Map<String, String> data) {
+		_data = data;
+	}
+
+	public void setDisabled(Boolean disabled) {
+		_disabled = disabled;
+	}
+
+	public void setHorizontalCard(HorizontalCard horizontalCard) {
+		_horizontalCard = horizontalCard;
+	}
+
+	public void setHref(String href) {
+		_href = href;
+	}
+
+	public void setIcon(String icon) {
+		_icon = icon;
+	}
+
+	public void setInteractive(Boolean interactive) {
+		_interactive = interactive;
+	}
+
+	public void setSelectable(Boolean selectable) {
+		_selectable = selectable;
+	}
+
+	public void setSelected(Boolean selected) {
+		_selected = selected;
+	}
+
+	public void setTitle(String title) {
+		_title = title;
+	}
+
+	@Override
+	protected void cleanUp() {
+		super.cleanUp();
+
+		_actionDropdownItems = null;
+		_data = null;
+		_disabled = null;
+		_horizontalCard = null;
+		_href = null;
+		_icon = null;
+		_interactive = null;
+		_selectable = null;
+		_selected = null;
+		_title = null;
+	}
+
+	@Override
+	protected String getHydratedModuleName() {
+		return "frontend-taglib-clay/HorizontalCard";
+	}
+
+	@Override
+	protected Map<String, Object> prepareProps(
+		Map<String, Object> initialProps) {
+
+		Map<String, Object> props = super.prepareProps(initialProps);
+
+		props.put("actions", getActionDropdownItems());
+		props.put("cssClass", processCssClasses(new LinkedHashSet<>()));
+		props.put("disabled", isDisabled());
+		props.put("href", getHref());
+		props.put("icon", getIcon());
+		props.put("id", getId());
+		props.put("interactive", getInteractive());
+		props.put("selectable", isSelectable());
+		props.put("selected", isSelected());
+		props.put("title", getTitle());
+
+		return props;
+	}
+
+	@Override
+	protected String processCssClasses(Set<String> cssClasses) {
+		if (isSelectable()) {
+			cssClasses.add("form-check");
+			cssClasses.add("form-check-card");
+			cssClasses.add("form-check-middle-left");
+		}
+		else {
+			cssClasses.add("card");
+			cssClasses.add("card-horizontal");
```

Comment:
Yeah, this one gave me a lot of headache :slightly_smiling_face:  

It can render in two different places, depending on `selectable`. If not `selectable`, it renders in container `div`. If `selectable`, it renders in content of `<ClayCheckbox>`, in root `dir` of the content.

Reply:
Could you please link where is that code located? I can't find it :S

Reply:
Do you mean "where is this class defined and used in Clay 3?" or "where is the code that renders it in final location after we set it here"?

I have a sneaking suspicions you mean former, in which case this is another case of 'see https://github.com/liferay-frontend/liferay-portal/pull/449#discussion_r512551932'. Clay 3 does not set it, [Clay 2 does](https://github.com/liferay/clay/blob/2.x/packages/clay-card/src/ClayHorizontalCard.soy#L32).

Reply:
That's why I couldn't find it then, yep xD.

Let's continue the conversation in https://github.com/liferay-frontend/liferay-portal/pull/449#discussion_r512551932 since it's the same case, I wouldn't add any cssClass nor markup specific to v2